packages/marble-api/openapis/marblecore-api.yaml

```diff
@@ -5214,6 +5336,75 @@ components:
           type: integer
         rate_limit_duration:
           type: integer
+    TestRunStatusDto:
+      type: string
+      enum: ['pending', 'up', 'down', 'unknown']
+    TestRunDto:
+      type: object
+      required:
+        - id
+        - scenario_id
+        - ref_iteration_id
+        - phantom_iteration_id
+        - start_date
+        - end_date
+        - creator_id
+        - status
+      properties:
+        id:
+          type: string
+          format: uuid
+        scenario_id:
+          type: string
+          format: uuid
+        ref_iteration_id:
+          type: string
+          format: uuid
+        phantom_iteration_id:
+          type: string
+          format: uuid
+        start_date:
+          type: string
+          format: date-time
+        end_date:
+          type: string
+          format: date-time
+        creator_id:
+          type: string
+          format: uuid
+        status:
+          $ref: '#/components/schemas/TestRunStatusDto'
+    TestRunCreateInputDto:
+      type: object
+      required:
+        - scenario_id
+        - ref_iteration_id
+        - phantom_iteration_id
+        - start_date
+        - end_date
+      properties:
+        scenario_id:
+          type: string
+          format: uuid
+        ref_iteration_id:
+          type: string
+          format: uuid
+        phantom_iteration_id:
+          type: string
+          format: uuid
+        start_date:
+          type: string
+          format: date-time
+        end_date:
+          type: string
+          format: date-time
```

Comment:
We can discuss on this DTO :
- `ref_iteration_id` seems unecessary with current business rules (= the only accepted version is the current live one) but we can still provide this in the payload and check this is the live_id on the backend (so in a future version, the API is ready to allow to pick a pair of version).
- `start_date` seems unecessary (unwanted)

Reply:
agreed on this - a test run should only be created on the currently live version (and cannot be created if none is live) - so we can remove this from the endpoint.
Similarly, I'd not allow to set a start date in the future - just use "now" (so no need to expose this either).
Finally, I'd go for just "scenario_iteration_id" (or if it's too ambiguous, "test_iteration_id" or such) instead of "phantom_iteration_id" : i find the concept of "phantom decisions" to be a convenient concept and naming to work with on the backend side, but IMO it's an implementation detail and I'd rather avoid "leaking" this naming into the rest of the product.